File: JavaCode/src/com/ysx/leetcode/medium/CustomSortString.java
package com.ysx.leetcode.medium;

/**
 * Leetcode: 791. Custom Sort String
 * https://leetcode.com/problems/custom-sort-string/description/
 */
public class CustomSortString {

    public static void main(String[] args) {
        CustomSortString test = new CustomSortString();
        String S = "cba";
        String T = "abcfde";
        String result = test.customSortString(S, T);
        System.out.println(result);
    }

    public String customSortString(String S, String T) {
        char[] arrayT = T.toCharArray();
        /**
         * 使用indexOf，进行冒泡排序
         */
        int length = arrayT.length;
        boolean flag = true;
        for (int i = 0; flag && i < length; i++) {
            flag = false;
            for (int j = 0; j < length - i - 1; j++) {
                if (S.indexOf(arrayT[j]) > S.indexOf(arrayT[j + 1])) {
                    char temp = arrayT[j];
                    arrayT[j] = arrayT[j + 1];
                    arrayT[j + 1] = temp;
                    flag = true;
                }
            }
        }
        return String.valueOf(arrayT);
    }
}

File: KotlinCode/src/leetcode/medium/CustomSortString.kt
package leetcode.medium

/**
 * Leetcode: 791. Custom Sort String
 * https://leetcode.com/problems/custom-sort-string/description/
 */
class Solution {
    fun customSortString(S: String, T: String): String {
        val arrayT = T.toCharArray()
        var flag : Boolean = true
        val size = arrayT.size
        for (i in 0 until size){
            if (!flag) {
                break
            }
            flag = false
            for (j in 0 until size - i - 1) {
                if (S.indexOf(arrayT[j]) > S.indexOf(arrayT[j + 1])) {
                    val temp:Char = arrayT[j]
                    arrayT[j] = arrayT[j + 1]
                    arrayT[j + 1] = temp
                    flag = true
                }
            }
        }
        return String(arrayT)
    }
}

fun main(args: Array<String>) {
    val test = Solution()
    val S = "cba"
    val T = "abcd"
    println(test.customSortString(S, T))
}
